src/oas2/guards.ts

```diff
@@ -21,3 +21,7 @@ export const isResponseObject = (maybeResponseObject: unknown): maybeResponseObj
     'schema' in maybeResponseObject ||
     'headers' in maybeResponseObject ||
     'examples' in maybeResponseObject);
+
+export function isValidScheme(scheme: unknown) {
+  return typeof scheme === 'string' && /^[a-zA-Z]\w+$/.test(scheme);
```

Comment:
Why don't we do a hard check against the possible values?

![image](https://user-images.githubusercontent.com/7423098/110133963-841b2200-7d92-11eb-9ade-0cdd4ae2c833.png)

Reply:
Good point, that would make it even safer